bin/beeline

```diff
@@ -23,8 +23,29 @@
 # Enter posix mode for bash
 set -o posix
 
+realpath () {
```

Comment:
Hi @srowen , yes the actual problem is to find to out where Spark is installed through symlink.

Yes, sourcing can follow the symlink, but the problem is how to find the path of sourcing file/symlink.

Say if we make `spark-shell` as a symlink and add to /usr/bin, when we execute this `spark-shell` in any directory, how to identify the path of sourcing file? If sourcing file is not linked to `/usr/bin/xx`, we still need to find out where Spark is installed.